### PR TITLE
add test_mssautoplot

### PR DESCRIPTION
**Pull request by HarshKhilawala**
@ReimarBauer PTAL! I have reduced it back to limit to imports.

### CODE REVIEW

File: tests/_test_utils/test_mssautoplot.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+"""
+
+    tests._test_utils.test_mssautoplot
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+    This module provides pytest functions to tests msui.msui
```
Comment (ReimarBauer) [inline tests/_test_utils/test_mssautoplot.py]: this line is for describing e.g. tests mssautoplot

File: tests/_test_utils/test_mssautoplot.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    This file is part of MSS.
+
+    :copyright: Copyright 2017 Joern Ungermann
```
Comment (ReimarBauer) [inline tests/_test_utils/test_mssautoplot.py]: there should be your name added and this year

File: tests/_test_utils/test_mssautoplot.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    :copyright: Copyright 2017-2022 by the MSS team, see AUTHORS.
```
Comment (ReimarBauer) [inline tests/_test_utils/test_mssautoplot.py]: 2023  by ...

File: tests/_test_utils/test_mssautoplot.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import mock
+import os
+import platform
+import argparse
+import pytest
+
+from mslib.utils.mssautoplot import load_from_ftml
+
+def test_load_from_ftml():
+
```
Comment (ReimarBauer) [inline tests/_test_utils/test_mssautoplot.py]: this should give access to the example.ftml in  PATH/tests/data/example.ftml

    sample_path = os.path.join(os.path.dirname(__file__), "..", "data")
    example_file = os.path.join(sample_path, "example.ftml")
    
    then use the function and verify it has the right data by comparing using assert

File: tests/_test_utils/test_mssautoplot.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+"""
+
+
+import sys
+import mock
```
Comment (ReimarBauer) [inline tests/_test_utils/test_mssautoplot.py]: clean to the only needed

File: tests/_test_utils/test_mssautoplot.py (hunk 1 above)
Comment (HarshKhilawala) [reply tests/_test_utils/test_mssautoplot.py]: Done!

File: tests/_test_utils/test_mssautoplot.py (hunk 2 above)
Comment (HarshKhilawala) [reply tests/_test_utils/test_mssautoplot.py]: Done!

File: tests/_test_utils/test_mssautoplot.py (hunk 3 above)
Comment (HarshKhilawala) [reply tests/_test_utils/test_mssautoplot.py]: Done!

**Comment by HarshKhilawala** [comment]
@ReimarBauer Next steps? (Also provide reference documentation for implementing steps) Thanks!

**Comment by ReimarBauer** [comment]
don't see a comparison for `assert load_from_ftml(example_file) == ?`

**Comment by ReimarBauer** [comment]
see test results

File: tests/_test_utils/test_mssautoplot.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+    tests._test_utils.test_mssautoplot
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+    This module provides pytest functions to tests mslib.utils.mssautoplot
+
+    This file is part of MSS.
+
+    :copyright: Copyright 2023 Harsh Khilawala
+    :copyright: Copyright 2017-2023 by the MSS team, see AUTHORS.
```
Comment (ReimarBauer) [inline tests/_test_utils/test_mssautoplot.py]: this module is new, 2023 only

**Comment by HarshKhilawala** [comment]
@ReimarBauer I just tried for load_from_ftml. Let me know if this is how it should be done. Now after writing test, how can we check the coverage for this `mssautoplot`?. so to know where and what lines to add testing.

File: tests/_test_utils/test_mssautoplot.py [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+import sys
+import mock
+import os
+import platform
+import argparse
+import pytest
+
+from mslib.utils.mssautoplot import load_from_ftml
+
+def test_load_from_ftml(filename):
```
Comment (ReimarBauer) [inline tests/_test_utils/test_mssautoplot.py]: please do something like this and use a file example :)

```
def test_load_from_ftml():
    sample_path = os.path.join(os.path.dirname(__file__), "..", "data")
    example_file = os.path.join(sample_path, "example.ftml")
```

**Comment by ReimarBauer** [comment]
you can do this for a specific test or for all. 

`pytest --cov-report html  tests/_test_utils/test_mssautoplot  --cov=mslib`

All tests the CI ran you can also try locally. They are defined in .github/workflows

**Comment by ReimarBauer** [comment]
a starter documentation: https://docs.pytest.org/en/7.1.x/getting-started.html

**Comment by HarshKhilawala** [comment]
@ReimarBauer I looked at test_airdata file to write tests myself and there is this ```@mock.patch("PyQt5.QtWidgets.QMessageBox.question", return_value=QtWidgets.QMessageBox.No)``` What does `@mock_patch` do? I don't know much about Qt yet but I am a quick-learner and can grasp important concept required to work on a project.

**Comment by ReimarBauer** [comment]
> @ReimarBauer I looked at test_airdata file to write tests myself and there is this `@mock.patch("PyQt5.QtWidgets.QMessageBox.question", return_value=QtWidgets.QMessageBox.No)` What does `@mock_patch` do? I don't know much about Qt yet but I am a quick-learner and can grasp important concept required to work on a project.


get first the test done for this function there is no mocking needed. 

with mocking you can replace functionalities like a Q/A with giving the answer instead of waiting for a user pressing a button. But for this function there is no mocking required.

**Comment by ReimarBauer** [comment]
@HarshKhilawala these lines I showed in the meet.
```
this should give access to the example.ftml in  PATH/tests/data/example.ftml

sample_path = os.path.join(os.path.dirname(__file__), "..", "data")
example_file = os.path.join(sample_path, "example.ftml")

then use the function and verify it has the right data by comparing using assert
```

**Comment by HarshKhilawala** [comment]
@ReimarBauer I tried to compare the exact return values for load_from_ftml but one of the values out of 2 variable was something like this ```[<mslib.msui.flighttrack.Waypoint object at 0x00000253D14E2FB0>, <mslib.msui.flighttrack.Waypoint object at 0x00000253D14E3C70>, <mslib.msui.flighttrack.Waypoint object at 0x00000253D14E39A0>, <mslib.msui.flighttrack.Waypoint object at 0x00000253D14E3C40>, <mslib.msui.flighttrack.Waypoint object at 0x00000253D1B26710>]``` (wp_list). So comparing exact value is not possible. Also fixed lint errors

**Comment by ReimarBauer** [comment]
How is it with this?


```
diff --git a/tests/_test_utils/test_mssautoplot.py b/tests/_test_utils/test_mssautoplot.py
index bb10c3fe..02e9db78 100644
--- a/tests/_test_utils/test_mssautoplot.py
+++ b/tests/_test_utils/test_mssautoplot.py
@@ -34,4 +34,12 @@ from mslib.utils.mssautoplot import load_from_ftml
 def test_load_from_ftml():
     sample_path = os.path.join(os.path.dirname(__file__), "..", "data")
     example_file = os.path.join(sample_path, "example.ftml")
-    assert load_from_ftml(example_file) is not None
+    data_list, wp_list = load_from_ftml(example_file)
+    assert data_list == [(55.15, -23.74, 0.0, 'B', 'Takeoff'),
+                         (42.99, -12.1, 350.0, 'A', ''),
+                         (52.785, -8.925, 380.0, 'Shannon', 'Dive'),
+                         (48.08, 11.28, 400.0, 'EDMO', ''),
+                         (63.74, 1.73, 0.0, 'C', 'Landing')
+                         ]
+    assert len(wp_list) == 5
+    assert type(wp_list[0]).__name__ == "Waypoint"
```


[test_mssautoplot.patch.txt](https://github.com/Open-MSS/MSS/files/10958462/test_mssautoplot.patch.txt)

you can apply it by patch -p1 < test_mssautoplot.patch.txt

**Review by ReimarBauer — CHANGES_REQUESTED** [review]
please have a look on the patch file

File: tests/_test_utils/test_mssautoplot.py (hunk 4 above)
Comment (HarshKhilawala) [reply tests/_test_utils/test_mssautoplot.py]: Done!

File: tests/_test_utils/test_mssautoplot.py (hunk 5 above)
Comment (HarshKhilawala) [reply tests/_test_utils/test_mssautoplot.py]: Done!

File: tests/_test_utils/test_mssautoplot.py (hunk 6 above)
Comment (HarshKhilawala) [reply tests/_test_utils/test_mssautoplot.py]: Done!

File: tests/_test_utils/test_mssautoplot.py (hunk 7 above)
Comment (HarshKhilawala) [reply tests/_test_utils/test_mssautoplot.py]: Done!

**Comment by HarshKhilawala** [comment]
@ReimarBauer Done! PTAL! I initially had thought of asserting for data_list but didn't consider matching its len and type.name 
Thanks!

**Comment by ReimarBauer** [comment]
@HarshKhilawala please look at the linter.

**Comment by HarshKhilawala** [comment]
@ReimarBauer How do I run tests locally which are in .github folder with yml extension?

**Comment by ReimarBauer** [comment]
flake8 tests 

after 
$ mamba install --file requirements.d/development.txt

**Comment by HarshKhilawala** [comment]
@ReimarBauer Done! I don't think that I might have access to re-run github workflows😅. GitHub should add a feature to allow contributors to re-run github workflows.

**Comment by ReimarBauer** [comment]
> @ReimarBauer Done! I don't think that I might have access to re-run github workflowssweat_smile. GitHub should add a feature to allow contributors to re-run github workflows.

github has kind of "spam" filter for its ressources implemented.  Before I have not merged a PR from a new contributor the tests won't start automated for a  PR. Next time after I merged this one it gets better :)

**Review by ReimarBauer — APPROVED** [review]
Thx

**Pull request merged**